crates/mfem/src/lib.rs

```diff
@@ -1,252 +1,1267 @@
-use cxx::memory::UniquePtrTarget;
-use cxx::{let_cxx_string, UniquePtr};
-use thiserror::Error;
+use autocxx::moveit::MakeCppStorage;
+use autocxx::prelude::*;
+use cxx::{let_cxx_string, memory::UniquePtrTarget};
+use paste::paste;
+use std::{
+    borrow::Cow, fmt, marker::PhantomData, path::Path, pin::Pin, ptr, slice,
+};
+
+/// *Pointers* to C++ objects are wrapped by this macro.
+macro_rules! wrap_mfem_sys {
+    ($(#[$doc: meta])* $name: ident <$($l: lifetime)?>) => {
```

Comment:
Do you think we might need to account for multiple lifetime parameters in here?

Reply:
On a second thougth, maybe that doesn't matter now, because the borrow checker will figure out that `&Mesh` and `&FEC` are passed in when constructing a `FES`, so those things must outlive it?

Reply:
The constructor [ produces a `FiniteElementSpaceBuilder<'a>`](https://github.com/Chris00/mfem-rs/blob/autocxx/crates/mfem/src/lib.rs#L1185) (and ultimately a [`FiniteElementSpace<'a>`](https://github.com/Chris00/mfem-rs/blob/autocxx/crates/mfem/src/lib.rs#L1275)) that depends on `Cow<'a, FiniteElementCollection>`.  The [function creating this](https://github.com/Chris00/mfem-rs/blob/autocxx/crates/mfem/src/lib.rs#L563) takes a `&'a mut Mesh` so it must outlive the mesh.  If the  `Cow<'a, FiniteElementCollection>` is also a reference (as opposed to an owned value), this value must also outlive `'a`.  As you know well, since we are dealing with C++ pointers, we must be careful about the lifetime we declare.  We may want to add tests that should not compile if we get the lifetimes wrong.